packages/core/src/client/hmr/index.ts

```diff
@@ -216,3 +172,78 @@ function tryApplyUpdates() {
     );
   }
 }
+
+let connection: WebSocket | null = null;
+const MAX_RETRIES = 100;
+
+// Establishing a WebSocket connection with the server.
+function connect() {
+  let isConnected = false;
+  connection = new WebSocket(socketUrl);
+
+  connection.onopen = function () {
+    isConnected = true;
+    if (typeof console !== 'undefined' && typeof console.info === 'function') {
+      // Notify users that the HMR has successfully connected.
+      console.info('[HMR] connected.');
+    }
+  };
+
+  // Unlike WebpackDevServer client, we won't try to reconnect
```

Comment:
This comment can be updated as Rsbuild will attempt to reconnect 😄

Reply:
OK😄